src/main/java/ru/spbstu/repository/VariantRepository.java

```diff
@@ -0,0 +1,33 @@
+package ru.spbstu.repository;
+
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import ru.spbstu.dto.VariantSearchRequest;
+import ru.spbstu.model.Variant;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.Optional;
+
+@Repository
+public interface VariantRepository extends CrudRepository<Variant, Long>, JpaSpecificationExecutor<Variant> {
+  default Optional<Variant> findVariant(VariantSearchRequest searchRequest) {
```

Comment:
I suggest not to use `VariantSearchRequest` in the repository layer.
Let's create use it in the service layer.
In the repository layer we can simply search by four separate fields. That would be enough.
In in the service layer (I think it can be implemented in the future task related to endpoint creation) we will introduce `VariantService` and there the repository method will be called.

Reply:
changed